src/main/scala/devices/uart/UART.scala

```diff
@@ -127,6 +150,24 @@ abstract class UART(busWidthBytes: Int, val c: UARTParams, divisorInit: Int = 0)
       RegField(c.divisorBits, div,
                  RegFieldDesc("div","Baud rate divisor",reset=Some(divisorInit))))
   )
+
+  val optionalparity = if (c.parity) Seq(
+    UARTCtrlRegs.parity -> RegFieldGroup("parity",Some("Odd/Even Parity Generation/Checking"),Seq(
```

Comment:
can the register group have a different name than the subfields? it is potentially confusing to use `parity` for both.